source/fake.go

```diff
@@ -45,18 +43,19 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-// NewFakeSource creates a new fakeSource with the given client and namespace scope.
-func NewFakeSource(dnsName string) (Source, error) {
-	if dnsName == "" {
-		dnsName = defaultDNSName
+// NewFakeSource creates a new fakeSource with the given config.
+func NewFakeSource(cfg *Config) (Source, error) {
```

Comment:
Interesting, I see it quite the opposite.

With the struct I can pass in whatever I care about only; I can do:
* `NewFakeSource(&Config{FQDNTemplate: "foo"})`
* `NewFakeSource(&Config{})` or even
* `NewFakeSource(nil)`

With possitional arguments I always have to define them even if I don't care and just want the default.

Furthermore, with every new initializable field I would need to modify the method signature of `NewFakeSource` breaking API compatibility.

Though, a shared struct for all sources isn't ideal either.